### DISCUSSION TITLE
what it is based on

### DISCUSSION THREAD

**Discussion by Ravid-Levy** [pymupdf/PyMuPDF#1798] [Q&A] [answered]
Hey,
how it can convert pdf to word

what it is based on?

which algorithms and which programs are the base of the mupdf?

is it support Hebrew language with rtl?

is pymupdf using swig?

**Comment by JorjMcKie** [comment, marked as answer]
Phew - 5 lines 5 questions. Here we go.

> how it can convert pdf to word

No direct support yet, but there is a document method in preparation for a future release. In the meantime please search PyPI: I remember a package doing exactly this using PyMuPDF.
You can also try to just open a PDF using a word editor like MS Word or LibreOffice. This sometimes also works ok.

> what it is based on?

What does that mean?

> which algorithms and which programs are the base of the mupdf?

Please look at https://github.com/ArtifexSoftware/mupdf

> is it support Hebrew language with rtl?

Hebrew can be read and written like any other language. What has rich text to do with that?

> is pymupdf using swig?

Yes
